notebooks/124-hugging-face-hub/124-hugging-face-hub.ipynb

```diff
@@ -0,0 +1,457 @@
+     "text": [
+      "Some weights of the model checkpoint at cardiffnlp/twitter-roberta-base-sentiment-latest were not used when initializing RobertaForSequenceClassification: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']\n",
+      "- This IS expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).\n",
+      "- This IS NOT expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n"
+     ]
+    }
+   ],
+   "source": [
+    "MODEL = \"cardiffnlp/twitter-roberta-base-sentiment-latest\"\n",
+    "\n",
+    "tokenizer = AutoTokenizer.from_pretrained(MODEL)\n",
+    "\n",
+    "model = AutoModelForSequenceClassification.from_pretrained(MODEL, torchscript=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Original Model inference\n",
+    "\n",
+    "Let's do a classification of a simple prompt below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1) positive 0.9485\n",
+      "2) neutral 0.0484\n",
+      "3) negative 0.0031\n"
+     ]
+    }
+   ],
+   "source": [
+    "text = \"HF models run perfectly with OpenVINO!\"\n",
+    "\n",
+    "encoded_input = tokenizer(text, return_tensors='pt')\n",
+    "output = model.forward(**encoded_input)\n",
+    "scores = output[0][0]\n",
+    "scores = torch.softmax(scores, dim=0).detach().numpy()\n",
+    "\n",
+    "def print_prediction(scores):\n",
+    "    for i, descending_index in enumerate(scores.argsort()[::-1]):\n",
+    "        label = model.config.id2label[descending_index]\n",
+    "        score = np.round(float(scores[descending_index]), 4)\n",
+    "        print(f\"{i+1}) {label} {score}\")\n",
+    "\n",
+    "print_prediction(scores)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Converting the Model to OpenVINO IR format\n",
+    "We use the OpenVINO [Model conversion API](https://docs.openvino.ai/2023.1/openvino_docs_model_processing_introduction.html#convert-a-model-in-python-convert-model) to convert the model (this one is implemented in PyTorch) to OpenVINO Intermediate Representation (IR).\n",
+    "\n",
+    "Note how we reuse our real `encoded_input`, passing it to the `convert_model` function. It will be used for model tracing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openvino as ov\n",
+    "\n",
+    "ov_model = ov.convert_model(model, example_input=dict(encoded_input))"
```

Comment:
Why didn't you add a model saving to avoid converting next time?

Reply:
did not think about it, as it is not a conversion tutorial but a model repo showcase. Will add anyway